v2/tools/generator/internal/config/group_configuration.go

```diff
@@ -161,6 +161,7 @@ func (gc *GroupConfiguration) UnmarshalYAML(value *yaml.Node) error {
 	}
 
 	gc.versions = make(map[string]*VersionConfiguration)
+	seenVersions := make(map[string]bool)
```

Comment:
I appreciate the thought that went into making a case-insensitive check, but this isn't needed. Our configuration is used for processing Swagger specs which may well contain both `Resource` and `resource` with different definitions, so our configuration must be case sensitive too.

It's sufficient to make a check against `gc.versions` to see if the version exists.

Also, please make the same change elsewhere - to check against the field containing the map with all the children, instead of maintaining a parallel local.